library/core/src/sync/atomic.rs

```diff
@@ -126,23 +126,13 @@ use crate::hint::spin_loop;
 
 /// Signals the processor that it is inside a busy-wait spin-loop ("spin lock").
 ///
-/// Upon receiving spin-loop signal the processor can optimize its behavior by, for example, saving
-/// power or switching hyper-threads.
-///
-/// This function is different from [`std::thread::yield_now`] which directly yields to the
-/// system's scheduler, whereas `spin_loop_hint` does not interact with the operating system.
-///
-/// A common use case for `spin_loop_hint` is implementing bounded optimistic spinning in a CAS
-/// loop in synchronization primitives. To avoid problems like priority inversion, it is strongly
-/// recommended that the spin loop is terminated after a finite amount of iterations and an
-/// appropriate blocking syscall is made.
+/// This function is expected to be deprecated in favor of
+/// [`core::hint::spin_loop`].
```

Comment:
Don’t mention crate name here, as the item will be re-exported in `libstd` and we want the docstring to make sense when viewed from documentation of either crate.

Reply:
So what do I do here? cc @jyn514 since he recently wanted me to link to `core::` instead of `crate::`.

Reply:
Make it a reference link instead of trying to put it directly in the text. This is the same thing I said in https://github.com/rust-lang/rust/pull/75945.

Reply:
The reason it had to be changed in https://github.com/rust-lang/rust/pull/76066#discussion_r480147557 is because you were using it in the link target, not the link text. This case is the link text.

Reply:
Wait, there is a difference between putting it as a link target and link text? I thought they are the same?

Reply:
https://www.markdownguide.org/basic-syntax/#reference-style-links

Reply:
I understand that but I thought you mean there is a difference when being use as link target or link text.

Reply:
Like this:
```suggestion
/// [`hint::spin_loop`].
```
and then below:
```
 /// [`hint::spin_loop`]: crate::hint::spin_loop
```

Reply:
Link _target_ is where you're linking _to_. Link _text_ is what _appears_ that you can click on. The target needs to be a valid Rust path; the text can be anything (in this case, a shortened version of the path).

Reply:
This was never fixed.

Reply:
Fixed.